spec-main/chromium-spec.ts

```diff
@@ -1089,7 +1089,11 @@ describe('chromium features', () => {
 
   describe('window.history', () => {
     describe('window.history.pushState', () => {
-      it('should push state after calling history.pushState() from the same url', (done) => {
+      it('should push state after calling history.pushState() from the same url', function (done) {
+        // In Electron the navigation history is maintained by listening to
+        // WebContents events, so window.history related tests are flaky.
+        this.retries(3)
```

Comment:
Is there some event we can wait for to have a deterministic test instead of retrying?

Reply:
There might be some events in navigation controller, I'll have a try.